src/polyfills.js

```diff
@@ -0,0 +1,33 @@
+// focus visible has very low support, so it's included unconditionally
+import 'focus-visible';
+
+function loadScript(path, cb) {
+  var s = document.createElement('script');
+  if (typeof cb === 'function') s.onload = cb;
+  s.src = path;
+  document.head.appendChild(s);
+}
+
+function loadPolyfills(cb) {
+  // load Promise first
+  if (!('Promise' in window)) {
+    loadScript(window.__PROMISE_POLYFILL__, () => {
+      loadOtherPolyfills().then(cb).catch(cb);
+    });
+  } else {
+    loadOtherPolyfills().then(cb).catch(cb);
+  }
+}
+
+function loadOtherPolyfills() {
+  const polyfills = [];
+
+  // add polyfills as needed here
+  if (!('Map' in window)) polyfills.push(import('core-js/features/map'));
+  if (!('Set' in window)) polyfills.push(import('core-js/features/set'));
+  if (!('Symbol' in window)) polyfills.push(import('core-js/features/symbol'));
+
+  return Promise.all(polyfills);
+}
+
+export default loadPolyfills;
```

Comment:
Adding polyfills seems to me like it's going beyond project configs.

Of course, if we want Carbon to be more than project configs, then I could see why polyfills would be handy.

Reply:
I think it is and it isn't. In my mind, this is a good example of why build setups are hard-many things are intertwined. A complete webpack config, I think, should include a polyfill bundling and loading strategy. That said, you could probably make an argument that webpack configuration is a different animal from simple static configs for things like linting or postcss. So maybe including webpack in this effort is going too far in itself?

I really don't know how I feel about it. I think I'd prefer to see a more complete setup here, but I'm open to reverting some of this work. @k80bowman do you have any thoughts on this?

I should add that this uses a completely dynamic loading strategy, but there are a couple other strategies that are worth considering. Based on the given project there are probably more pragmatic approaches.

Reply:
I like the idea of including polyfills, but I also like the idea of Carbon being a complete setup rather than only a library of configs. 

As far as alternative strategies go, however, is there a reason you are using a dynamic loading strategy here?

Reply:
dynamic loading is all in the name of performance. Alternative approaches require shipping unneeded code to users, but can be simpler for developers. 🤔 It could be something we skip for now and discuss further.